src/backtracking/knight_tour.rs

```diff
@@ -0,0 +1,188 @@
+//! This module contains the implementation of the Knight's Tour problem.
+//!
+//! The Knight's Tour is a classic chess problem where the objective is to move a knight to every square on a chessboard exactly once.
+
+/// Finds the Knight's Tour starting from the specified position.
+///
+/// # Arguments
+///
+/// * `size_x` - The width of the chessboard.
+/// * `size_y` - The height of the chessboard.
+/// * `start_x` - The x-coordinate of the starting position.
+/// * `start_y` - The y-coordinate of the starting position.
+///
+/// # Returns
+///
+/// A tour matrix if the tour was found or None if not found.
+/// The tour matrix returned is essentially the board field of the `KnightTour`
+/// struct `Vec<Vec<i32>>`. It represents the sequence of moves made by the
+/// knight on the chessboard, with each cell containing the order in which the knight visited that square.
+pub fn find_knight_tour(
+    size_x: usize,
+    size_y: usize,
+    start_x: usize,
+    start_y: usize,
+) -> Option<Vec<Vec<usize>>> {
+    let mut tour = KnightTour::new(size_x, size_y);
+    tour.find_tour(start_x, start_y)
+}
+
+/// Represents the KnightTour struct which implements the Knight's Tour problem.
+struct KnightTour {
+    board_size_x: usize,
+    board_size_y: usize,
```

Comment:
These two fields are not needed. Both sizes could be queried using the `board`.

Reply:
Solved.